### PR TITLE
Inital start to be able to redirect from http to https.

### PR DESCRIPTION
**Pull request by cw-aleks**
See issue #5946 
https://github.com/openshift/origin/issues/5946

### CODE REVIEW

**Comment by danmcp** [comment]
@ramr @knobunc @openshift/api-review Review please.

File: pkg/route/api/types.go [hunk 1]
```diff
@@ -133,20 +133,31 @@ type TLSConfig struct {
...
+        // DEPRECATED: Keep this for backward compatibility
 	// InsecureEdgeTerminationPolicy indicates the desired behavior for
 	// insecure connections to an edge-terminated route:
 	//   disable, allow or redirect
 	InsecureEdgeTerminationPolicy InsecureEdgeTerminationPolicyType
+
+	// InsecureTerminationPolicy indicates the desired behavior for
+	// insecure connections to one of the tls (edge,passthrough,reencrypt) route:
+	//   disable, allow or redirect
+	InsecureTerminationPolicy InsecureTerminationPolicyType
```
Comment (smarterclayton) [inline pkg/route/api/types.go]: Why would we not just add "Redirect" to "InsecureEdgeTerminationPolicy"?  I don't see the value in adding the new external field - we can just rename it in a future version.  Broadening EdgeTerminationPolicy to apply to all termination types doesn't seem to have a disadvantage I can see.  Examples?

Comment (knobunc) [inline pkg/route/api/types.go]: https://github.com/openshift/origin/pull/11953 is open to do just that.

**Comment by knobunc** [comment]
@cw-aleks: Sorry, I missed this until today.  I am closing in preference to https://github.com/openshift/origin/pull/11953

Please tell me if that PR does not accommodate your needs.

**Pull request closed without merging**